### PR TITLE
feat!: disable suspense by default, add suspense hooks

### PR DESCRIPTION
**Pull request by toddbaert**
This PR:

- inverts the suspense defaults (we now do not suspend by default, you have to add `suspend:true` in options)
- adds `useSuspenseFlag` (analogous to `useSuspenseXxx`) in other libraries, which behaves the same as `useFlag` with `{ suspend: true }`
- updates README (specifically encourages use use "query-style" hooks over type-specific hooks
- adds `@experimental` jsdoc marker to all suspense options and hooks
- associated tests

Things to consider:

- I did not add `useSuspense{Type}FlagValue` and `useSuspense{Type}FlagDetails` hooks; we could do this if we wanted, but IMO these are already not the primary APIs we want to push users toward in react - we want them to use the generic `useFlag` and `useSuspenseFlag` which return the react query interfaces.

Fixes: https://github.com/open-feature/js-sdk/issues/933

### CODE REVIEW

File: packages/react/README.md [hunk 1]
```diff
@@ -216,20 +220,29 @@ You can disable this feature in the hook options (or in the [OpenFeatureProvider
 
 ```tsx
 function Page() {
-  const showNewMessage = useBooleanFlagValue('new-message', false, { updateOnConfigurationChanged: false });
+  const { value: showNewMessage } = useFlag('new-message', false, { updateOnConfigurationChanged: false });
```
Comment (toddbaert) [inline packages/react/README.md:223]: updated the README to push people toward the query-style API.

File: packages/react/src/evaluation/use-feature-flag.ts [hunk 2]
```diff
@@ -31,6 +31,9 @@ type ConstrainedFlagQuery<T> = FlagQuery<
           : JsonValue
 >;
 
+// suspense options removed for the useSuspenseFlag hooks
```
Comment (toddbaert) [inline packages/react/src/evaluation/use-feature-flag.ts:34]: Remove suspense from the options of the `useSuspenseFlag` hook (it's always on).

File: packages/react/src/evaluation/use-feature-flag.ts [hunk 3]
```diff
@@ -70,6 +73,26 @@ T extends boolean
   return query as unknown as ConstrainedFlagQuery<T>;
 }
 
+// alias to the return value of useFlag, used to keep useSuspenseFlag consistent
+type ReturnTypeOfUseFlag<T extends FlagValue> = ReturnType<typeof useFlag<T>>
```
Comment (toddbaert) [inline packages/react/src/evaluation/use-feature-flag.ts:77]: Aliased this type so `useFlag` and `useSuspense` flag stay synced up.

File: packages/react/test/evaluation.spec.tsx [hunk 4]
```diff
@@ -254,14 +255,13 @@ describe('evaluation', () => {
      */
     const DELAY = 100;
 
-    const SUSPENSE_ON = 'suspense';
```
Comment (toddbaert) [inline packages/react/test/evaluation.spec.tsx:257]: I improved the isolation of the test suites a bit.

File: packages/react/test/evaluation.spec.tsx [hunk 5]
```diff
@@ -454,48 +470,119 @@ describe('evaluation', () => {
...
+        await waitFor(() => expect(screen.queryByText(TARGETED_FLAG_VALUE)).toBeInTheDocument(), {
+          timeout: DELAY * 2,
+        });
+      });
     });
 
-    it('should show the previous UI while reconciling during a transition', async () => {
-      await OpenFeature.setContext(SUSPENSE_OFF, {});
-      OpenFeature.setProvider(SUSPENSE_ON, suspendingProvider());
+    describe.each([useFlag, useSuspenseFlag])('suspend=true', () => {
```
Comment (toddbaert) [inline packages/react/test/evaluation.spec.tsx:512]: Note this runs multiple times via `.each`.

File: packages/react/src/evaluation/use-feature-flag.ts (hunk 2 above)
Comment (beeme1mr) [reply packages/react/src/evaluation/use-feature-flag.ts:34]: Could the `suspendWhileReconciling` still be useful to control?

Comment (toddbaert) [reply packages/react/src/evaluation/use-feature-flag.ts:34]: I think it's best to make the suspense hooks enable all suspense options by default. I think if more fine grained control is needed I'd use the 'useFlag' hook. I think in most cases , you either want all suspense or none.

File: packages/react/README.md [hunk 6]
```diff
@@ -199,10 +199,14 @@ You can disable this feature in the hook options (or in the [OpenFeatureProvider
...
-    <MyComponents></MyComponents>
-  )
+    <div className="App">
+      <header className="App-header">
+        {showNewMessage ? <p>Welcome to this OpenFeature-enabled React app!</p> : <p>Welcome to this React app.</p>}
+      </header>
+    </div>
+  );
 }
 ```
```
Comment (beeme1mr) [inline packages/react/README.md:212]: You could link the static context paradigm concepts page now if you want.
https://openfeature.dev/docs/reference/concepts/sdk-paradigms

**Review by beeme1mr — APPROVED** [review]

File: packages/react/src/evaluation/use-feature-flag.ts (hunk 3 above)
Comment (lukas-reining) [reply packages/react/src/evaluation/use-feature-flag.ts:77]: Just a naming thing but I would expect a type name like `UseFlagReturn` or even `UseFlagResult`.

**Review by lukas-reining — APPROVED** [review]
Change looks good to me, left a thought on return type naming.

File: packages/react/src/evaluation/use-feature-flag.ts (hunk 3 above)
Comment (toddbaert) [inline packages/react/src/evaluation/use-feature-flag.ts:77]:
```suggestion
type UseFlagReturn<T extends FlagValue> = ReturnType<typeof useFlag<T>>
```

File: packages/react/src/evaluation/use-feature-flag.ts [hunk 7]
```diff
@@ -70,6 +73,26 @@ T extends boolean
...
+
+/**
+ * Equivalent to {@link useFlag} with `options: { suspend: true }`
+ * @experimental Suspense is an experimental feature subject to change in future versions.
+ * @param {string} flagKey the flag identifier
+ * @template {FlagValue} T A optional generic argument constraining the default.
+ * @param {T} defaultValue the default value; used to determine what resolved type should be used.
+ * @param {NoSuspenseOptions} options for this evaluation
+ * @returns { ReturnTypeOfUseFlag<T> } a queryable object containing useful information about the flag.
```
Comment (toddbaert) [inline packages/react/src/evaluation/use-feature-flag.ts:86]:
```suggestion
 * @returns { UseFlagReturn<T> } a queryable object containing useful information about the flag.
```

File: packages/react/src/evaluation/use-feature-flag.ts [hunk 8]
```diff
@@ -70,6 +73,26 @@ T extends boolean
...
+ */
+export function useSuspenseFlag<T extends FlagValue = FlagValue>(
+  flagKey: string,
+  defaultValue: T,
+  options?: NoSuspenseOptions,
+): ReturnTypeOfUseFlag<T> {
```
Comment (toddbaert) [inline packages/react/src/evaluation/use-feature-flag.ts:92]:
```suggestion
): UseFlagReturn<T> {
```

**Comment by toddbaert** [comment]
@tjosepo Just FYI b/c you did some recent related work. I plan on merging/releasing this EOD.

**Pull request merged**